src/Oliver.Client/Executing/Executor.cs

```diff
@@ -59,12 +63,21 @@ private async Task Execute(Configurations.Client.Instance instance, Execution ex
                 var logs = new List<string> { $"Starting step {step.Order}: '{step.Name}'" };
 
                 var command = Substitute(step.Command, variables?.Values);
-                var folder = Substitute(step.WorkingFolder, variables?.Values);
+
+                var folder = Substitute(step.WorkingFolder ?? "", variables?.Values);
                 folder = Path.GetFullPath(folder, Path.GetFullPath(this.instanceOptions?.Value?.DefaultFolder ?? "."));
+                if (!Directory.Exists(folder))
+                {
+                    Directory.CreateDirectory(folder);
+                    logs.Add($"Folder: '{folder}' created.");
+                }
                 logs.Add($"Executing at folder: '{folder}'");
 
+                Common.Models.File file;
                 var result = step.Type switch
                 {
+                    Template.StepType.Archive when (file = await GetArchive(step.FileName)) is { }
```

Comment:
Am I right that if you couldn't find the file metadata, you would throw `NotImplementedException`?

Reply:
yes

Reply:
yeah, need to split notimplemented type and notfound file

Reply:
done